contentmgr/gc.go

```diff
@@ -147,6 +146,27 @@ func (cm *ContentManager) UnpinContent(ctx context.Context, contid uint) error {
 		return err
 	}
 
+	if pin.AggregatedIn > 0 {
```

Comment:
@neelvirdy move this logic down, it should be done after the blocks are removed

Reply:
i moved all contents table updates below blocks are removed, and put them in one transaction. lmk if you were thinking of something else. i think doing all the unpinning DB updates in one transaction then deleting blocks would be a bit cleaner so the end user sees the delete happen and we can always GC the deleted blocks later if that ends up failing. but would consider that for a separate PR